WebDriverAgentLib/Categories/XCUIElement+FBTyping.m

```diff
@@ -13,15 +13,42 @@
 #import "FBKeyboard.h"
 #import "NSString+FBVisualLength.h"
 #import "XCUIElement+FBTap.h"
+#import "XCUIElement+FBUtilities.h"
 
 @implementation XCUIElement (FBTyping)
 
+- (BOOL)fb_prepareForTextInputWithError:(NSError **)error
+{
+  BOOL isKeyboardAlredyVisible = [FBKeyboard waitUntilVisibleForApplication:self.application timeout:-1 error:error];
+  if (isKeyboardAlredyVisible && self.hasKeyboardFocus) {
+    return YES;
+  }
+  
+  int retriesCount = 2;
+  int tryNum = 0;
+  while (tryNum < retriesCount) {
```

Comment:
Does Objective-C not have a `for` loop that does this count/tries handling for you?

Reply:
what do you mean? any example in JS?

Reply:
Just a normal C-style `for` loop (`for (int i = 0; i < sizeof(something); i++) { ... }`)

Reply:
changed to `for`